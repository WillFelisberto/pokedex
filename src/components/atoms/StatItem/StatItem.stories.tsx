import type { Meta, StoryObj } from '@storybook/react';

import { StatItem } from './StatItem';

const meta: Meta<typeof StatItem> = {
  title: 'Atoms/StatItem',
  component: StatItem,
  tags: ['autodocs'],

  argTypes: {
    value: { control: 'text' },
    type: {
      control: 'select',
      options: ['height', 'weight', 'category', 'ability'],
      description: 'Defines the type of stat, which determines the icon displayed.'
    }
  }
};

export default meta;

type Story = StoryObj<typeof StatItem>;

export const Default: Story = {
  args: {
    value: '9.0 kg',
    type: 'weight'
  }
};
