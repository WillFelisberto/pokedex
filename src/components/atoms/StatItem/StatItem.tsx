import clsx from 'clsx';
import Image from 'next/image';
import React from 'react';

interface StatItemProps {
  value: string | number;
  type: 'height' | 'weight' | 'category' | 'ability';
  className?: string;
}

export const StatItem: React.FC<StatItemProps> = ({ value, type, className }) => {
  const labelText = {
    height: 'Altura',
    weight: 'Peso',
    category: 'Categoria',
    ability: 'Habilidade'
  };

  return (
    <div className={clsx('flex flex-col gap-[4.5px] items-left  p-3 rounded-lg  ', className)}>
      <div className="flex items-end gap-[8.5px] text-gray-500 text-xs font-semibold uppercase">
        <Image src={`/icons/svg/outlined/${type}.svg`} alt={type} width={18} height={18} />
        <span>{labelText[type] || type}</span>
      </div>
      <div className="mt-1 font-(--font-poppins)  w-full py-[8px] bg-white rounded-[15px] text-gray-800 font-bold text-lg text-center border border-[#E5E5E5]">
        {value}
      </div>
    </div>
  );
};
