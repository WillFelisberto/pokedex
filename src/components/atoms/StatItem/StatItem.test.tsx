import { render, screen } from '@/tests/test-utils';

import { StatItem } from './StatItem';

describe('<StatItem />', () => {
  it('should render the correct label for height', () => {
    const { container } = render(<StatItem type="height" value="0.5 m" />);
    expect(screen.getByText(/Altura/i)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render the correct label for weight', () => {
    render(<StatItem type="weight" value="9.0 kg" />);
    expect(screen.getByText(/Peso/i)).toBeInTheDocument();
  });

  it('should render the correct label for category', () => {
    render(<StatItem type="category" value="Tiny Turtle" />);
    expect(screen.getByText(/Categoria/i)).toBeInTheDocument();
  });

  it('should render the correct label for ability', () => {
    render(<StatItem type="ability" value="Torrent" />);
    expect(screen.getByText(/Habilidade/i)).toBeInTheDocument();
  });

  it('should display the correct value', () => {
    render(<StatItem type="height" value="0.5 m" />);
    expect(screen.getByText(/0.5 m/i)).toBeInTheDocument();
  });

  it('should render the correct icon for type', () => {
    render(<StatItem type="height" value="0.5 m" />);
    const icon = screen.getByAltText('height');
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveAttribute('src', '/icons/svg/outlined/height.svg');
  });
});
