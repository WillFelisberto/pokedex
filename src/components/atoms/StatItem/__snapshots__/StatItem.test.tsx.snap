// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatItem /> should render the correct label for height 1`] = `
<div>
  <div
    class="flex flex-col gap-[4.5px] items-left   rounded-lg  "
  >
    <div
      class="flex items-end gap-[8.5px] h-[16px] text-gray-500 text-xs font-semibold uppercase"
    >
      <img
        alt="height"
        height="15"
        src="/icons/svg/outlined/height.svg"
        width="15"
      />
      <span>
        Height
      </span>
    </div>
    <div
      class="mt-1 font-(--font-poppins) h-[42px]  w-full py-[8px] bg-white rounded-[15px] text-gray-800 font-bold text-lg text-center border border-[#E5E5E5]"
    >
      0.5 m
    </div>
  </div>
</div>
`;
